### ISSUE TITLE
Timeline show one date from all years

### ISSUE THREAD

**Issue by savegame** [photoview/photoview#632]
**Describe the bug**
First i add 2017 year to indexing, than 2018 ... Than when i try view in time line dates like "from 2018 and earler", its show 2017 dates first

**To Reproduce**
1. First add first album from one month or year ( as in my case, first i add 2017 year for indexing )
2. Then add next album ( in my case its 2018 year ) 
3. Open timeline, and set up "from 2018 and earler"
4. It still show first 2017 year, and if it has same dates as in 2018, it show both pictures in one line from both years

**Expected behavior**
Should show 2018 first, then 2017

**Screenshots**

![изображение](https://user-images.githubusercontent.com/16311332/150502795-f421184a-903c-4ce8-a8e6-6ed3d7846701.png)  
![изображение](https://user-images.githubusercontent.com/16311332/150502356-458691b2-a09a-4892-bf8b-bb42bf10a390.png)  

**Your environment:**
Feel free to remove and add what you think might be useful.

 - Build it in jail in FreeBSD 12.2
 - Database: sqlite ( just for testing this software, later i'll try with postgresql ) 
 - Browser: Firefox 96.0 ( Ubuntu 21.10 )
 - Version: 2.3.9

**Additional context**
Host system is TrueNAS, in jail ( FreeBSD 12.3 )

**Comment by savegame** [comment]
P.S. in my previous case, i use photoview in this algorithm (with SQLite): 
1. add folder for scanning /media/2017/01  ( for testing ) 
2. delete this album in settings
3. add this album again in settings and press scan
4. delete it again in settings
5. add /media/2017 as album and press scan
6. add /media/2018 as album and press scan 
- after than i was missing 01 folder from 2017 year ( it shown as empty ) , and bug, described upper

**Comment by savegame** [comment]
@viktorstrate looks like with postgres bug is not reproduced ( later i'll try reproduce this bug on separate jail, just for test purposes (with SQLite again))
and i have another bug =)   
looks like in inforation panel, path to pictures album has wrong format 
![изображение](https://user-images.githubusercontent.com/16311332/152947543-96a5e9c5-89ef-478c-8926-efb047785bbb.png)   
real album path is   
![изображение](https://user-images.githubusercontent.com/16311332/152947795-0cca672a-3c95-491b-bed5-f4e07a54048f.png)

**Comment by viktorstrate** [comment]
Right, thank you for looking into this.
This seems to be related to sqlite then. Sqlite is very unstable see #618 for example.
So I don't think we can move forward with this issue until the latter is resolved.
Until that's fixed I would discourage the use of sqlite and use Mysql or Postgres instead.

As for the date, I just improved the scanners ability to pick up the correct date from the metadata in the media files.
But the scanner doesn't look at the directory path to determine this, and It might be hard to determine since the day and month can be interchanged depending on language[^1].


[^1]: https://en.wikipedia.org/wiki/Date_format_by_country

**Comment by savegame** [comment]
but i am talking about album path. not about date ( in my case. album path - same as date, because my photos are sorted b script to subfolders ) 
but, in info panel, it show album path like: `/12/2017/21` , but in real it is in `/2017/12/21`, in `albums` tab all paths correct, as in filesystem, in `about picture (or picture info)` panel album path is different  
![изображение](https://user-images.githubusercontent.com/16311332/152983917-0bf861df-ef18-42b4-a462-749dd45bd790.png)

**Comment by viktorstrate** [comment]
Ahh I see. I've quickly fixed that in a43f29e

**Comment by savegame** [comment]
thanks ) 
can i ask some another question?  ) 

how i can use static site, built with `npm run build` ? 
should i use `PHOTOVIEW_SERVE_UI=1` in `api/.env`? maybe something else?

**Comment by viktorstrate** [comment]
That's the easiest way. Take a loot at https://photoview.github.io/docs/installation-manual/#configure-photoview
Otherwise if you have further questions regarding setup, I'd suggets you ask in the Discord channel, then it's much easier to chat about.

**Comment by savegame** [comment]
@viktorstrate i'll check your latest `master` branch later, and post here if there's anything wrong with the album path in the `photo info` panel

**Comment by fierceX** [comment]
@savegame Hello, I also want to deploy on truenas, but I am not familiar with freebsd. How do you deploy it? can you tell me something about it? Thank you

**Comment by savegame** [comment]
@fierceX Hello, dont have enough time, still not test your lats commit, about "path in info panel", sorry, i'll try it today :)
On FreeNAS i just install all dependencies, then download your repo and build it as on any Linux, nothing special. Its all in jail container.

**Comment by mih** [comment]
I am running into the same issue (recent code, also sqlite). I looked for the timestamp format in the `media` table stored in the sqlite DB. Here is an example record:

```json
{
  "id": 7611,
  "created_at": "2024-05-25 04:44:52.485521161+02:00",
  "updated_at": "2024-05-25 04:46:23.127583467+02:00",
  "title": "IMG_20190215_155633.jpg",
  "path": "/home/photos/<cut>/IMG_20190215_155633.jpg",
  "path_hash": "49243303550a7e683fadbe3bbcac2c72",
  "album_id": 94,
  "exif_id": 7607,
  "date_shot": "2019-02-15 15:56:33+00:00",
  "type": "photo",
  "video_metadata_id": null,
  "side_car_path": null,
  "side_car_hash": null,
  "blurhash": "LVDAGyVCs.V?Knn}V?ngtoRjs.s*"
}
```

Without having a comparison how a record would look with postgres, I'd speculate that these are standard timestamps.

If true, this would indicate the cause being the read/interpretation of the timestamp, and thereby somewhat unrelated to any sqlite write-access issues.

**Comment by kkovaletp** [comment]
@mih, thanks for your additional info. It looks like this issue needs some deeper investigation

**Comment by mih** [comment]
Looking further, when I navigate to the timeline view, the relevant graphql query request has the following payload:

```json
{
  "operationName": "myTimeline",
  "variables": {
    "onlyFavorites": false,
    "offset": 0,
    "limit": 200
  },
  "query": "query myTimeline($onlyFavorites: Boolean, $limit: Int, $offset: Int, $fromDate: Time) {\n  myTimeline(\n    onlyFavorites: $onlyFavorites\n    fromDate: $fromDate\n    paginate: {limit: $limit, offset: $offset}\n  ) {\n    id\n    title\n    type\n    blurhash\n    thumbnail {\n      url\n      width\n      height\n      __typename\n    }\n    highRes {\n      url\n      width\n      height\n      __typename\n    }\n    videoWeb {\n      url\n      __typename\n    }\n    favorite\n    album {\n      id\n      title\n      __typename\n    }\n    date\n    __typename\n  }\n}\n"
}
```

The pretty-printed query is

```graphql
query myTimeline($onlyFavorites: Boolean, $limit: Int, $offset: Int, $fromDate: Time) {
  myTimeline(
    onlyFavorites: $onlyFavorites
    fromDate: $fromDate
    paginate: {limit: $limit, offset: $offset}
  ) {
    id
    title
    type
    blurhash
    thumbnail {
      url
      width
      height
      __typename
    }
    highRes {
      url
      width
      height
      __typename
    }
    videoWeb {
      url
      __typename
    }
    favorite
    album {
      id
      title
      __typename
    }
    date
    __typename
  }
}
```

The structure of the graphql query response is

```json
{
  "data": {
    "myTimeline": [
      {
        "id": "9979",
        "title": "...",
        "type": "Photo",
        "blurhash": "LICimyi^NrV[~UV?RkkC^+%1M_o#",
        "thumbnail": {
          "url": "/api/photo/thumbnail_....jpg",
          "width": 1024,
          "height": 561,
          "__typename": "MediaURL"
        },
        "highRes": {
          "url": "/api/photo/....jpg",
          "width": 3144,
          "height": 1725,
          "__typename": "MediaURL"
        },
        "videoWeb": null,
        "favorite": false,
        "album": {
          "id": "101",
          "title": "...",
          "__typename": "Album"
        },
        "date": "2010-12-31T21:22:17Z",
        "__typename": "Media"
      },
...
```

Importantly, the order in the response *is the problematic one* (via `jq '.data.myTimeline[].date' < response.json`):

```
"2010-12-31T21:22:17Z"
"2009-12-31T21:03:54Z"
"2009-12-31T21:02:59Z"
"2009-12-31T21:01:07Z"
"2009-12-31T21:00:22Z"
"2010-12-31T19:30:27Z"
"2022-12-30T12:18:11Z"
"2009-12-30T19:11:24Z"
"2009-12-30T18:46:04Z"
"2009-12-30T18:45:51Z"
"2010-12-30T17:13:30Z"
"2010-12-30T16:57:01Z"
"2010-12-30T16:29:38Z"
"2010-12-30T16:11:11Z"
"2010-12-30T16:10:48Z"
"2010-12-30T11:52:25Z"
"2010-12-30T11:51:42Z"
"2010-12-30T11:47:04Z"
"2010-12-30T11:45:54Z"
"2010-12-29T17:10:33Z"
"2010-12-29T16:03:27Z"
"2010-12-29T15:32:18Z"
"2010-12-29T14:53:06Z"
"2010-12-29T14:30:16Z"
"2010-12-29T14:29:33Z"
"2022-12-28T19:02:18Z"
"2022-12-28T19:02:12Z"
"2022-12-28T15:29:29Z"
"2022-12-28T15:29:18Z"
...
```

This seems to hint at the implementation of the query (post)processing being the culprit here.

**Comment by kkovaletp** [comment]
There is another issue, related to the ordering for SQLite DB: #954. Please double-check whether it is the same as you just reported.
If that is the case, do you think that once that PR is merged, this issue will be fixed completely? If yes, I'll link them together)

**Comment by mih** [comment]
I just replied over there: That patch fixes this issue for me. Thanks for pointing it out!
